Userland/Libraries/LibGfx/Font/FontDatabase.cpp

```diff
@@ -121,54 +121,53 @@ struct FontDatabase::Private {
     HashMap<FlyString, Vector<NonnullRefPtr<Typeface>>, AK::ASCIICaseInsensitiveFlyStringTraits> typefaces;
 };
 
-void FontDatabase::load_all_fonts_from_path(DeprecatedString const& root)
+void FontDatabase::load_all_fonts_from_path(DeprecatedString const& path)
 {
-    Queue<DeprecatedString> path_queue;
-    path_queue.enqueue(root);
-
-    while (!path_queue.is_empty()) {
-        auto current_directory = path_queue.dequeue();
-        Core::DirIterator dir_iterator(current_directory, Core::DirIterator::SkipParentAndBaseDir);
-        if (dir_iterator.has_error()) {
-            dbgln("FontDatabase::load_all_fonts_from_path('{}'): {}", root, dir_iterator.error());
-            continue;
-        }
-        while (dir_iterator.has_next()) {
-            auto path = dir_iterator.next_full_path();
+    load_all_fonts_from_uri(DeprecatedString::formatted("file://{}", path));
+}
 
-            if (FileSystem::is_directory(path)) {
-                path_queue.enqueue(path);
-                continue;
+void FontDatabase::load_all_fonts_from_uri(StringView uri)
+{
+    auto root_or_error = Core::Resource::load_from_uri(uri);
+    if (root_or_error.is_error()) {
+        dbgln("FontDatabase::load_all_fonts_from_uri('{}'): {}", uri, root_or_error.error());
+        return;
+    }
+    auto root = root_or_error.release_value();
+
+    root->for_each_descendant_file([this](Core::Resource const& resource) -> IterationDecision {
+        // FIXME: Use Resources and their bytes/streams throughout so we don't have to use the path here
+        auto path_string = resource.filesystem_path().release_value();
+        auto path = path_string.bytes_as_string_view();
+        if (path.ends_with(".font"sv)) {
```

Comment:
Files named just `.font` shouldn't be considered, so using LexicalPath for the `resource.filesystem_path()` will allow the use of `extension()` which is more correct.

Reply:
This is a preexsiting issue, but yeah i can drop a fix in a sep commit.